onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -617,17 +617,29 @@ def _get_data(self, query, fields, sort, start, limit, is_public_request):
         self.set_object_list(
             query, fields, sort, start, limit, is_public_request)
 
+        retrieval_threshold = getattr(
+            settings, "SUBMISSION_RETRIEVAL_THRESHOLD", 10000)
         pagination_keys = [self.paginator.page_query_param,
                            self.paginator.page_size_query_param]
         query_param_keys = self.request.query_params
         should_paginate = any([k in query_param_keys for k in pagination_keys])
+
+        if not should_paginate and not is_public_request:
```

Comment:
Does this mean we will not paginate public forms that exceed the threshold?

Reply:
At the moment yes; didn't think of this while implementing the feature... Any thoughts/reservations with always paginating public requests? _Seems a bit painful to calculate the entire count(getting all public XForms and summing the `num_of_submissions` of public data that's currently available..._

Reply:
This is for a single form request at a time, not all public forms at the same. Also, we could go with `num_of_submissions` that is not calculated on the fly, it is a safe estimate in most situations.

Reply:
I withdraw my statement, I see that is_public_request represents access to the `/api/v1/data/public` and not to one single form data.